src/Aspire.Dashboard/Telemetry/AspireTelemetryService.cs

```diff
@@ -0,0 +1,364 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using Aspire.Dashboard.Configuration;
+using Aspire.Dashboard.Utils;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Dashboard.Telemetry;
+
+public sealed class AspireTelemetryService(
+    IOptions<DashboardOptions> options,
+    ILocalStorage localStorage,
+    AspireTelemetryService.ITelemetrySender telemetrySender,
+    ILogger<AspireTelemetryService> logger) : IAspireTelemetryService
+{
+    private readonly Lazy<HttpClient?> _httpClient = new(() => CreateHttpClient(options.Value.DebugSession));
+    private bool? _telemetrySupported;
+    private bool? _telemetryEnabled;
+
+    public bool IsTelemetryInitialized => _telemetrySupported is not null && _telemetryEnabled is not null;
+    public bool IsTelemetrySupported => _telemetrySupported ?? throw new ArgumentNullException(nameof(_telemetrySupported), "InitializeAsync has not been called yet");
+    public bool IsTelemetryEnabled => _telemetryEnabled ?? throw new ArgumentNullException(nameof(_telemetryEnabled), "InitializeAsync has not been called yet");
+
+    public async Task InitializeAsync()
+    {
+        logger.LogDebug("Initializing telemetry service.");
+
+        if (_telemetrySupported is not null)
+        {
+            return;
+        }
+
+        _telemetrySupported = await GetTelemetrySupportedAsync(_httpClient.Value, telemetrySender, logger).ConfigureAwait(false);
```

Comment:
HttpClient should be moved inside telemetrySender. The telemetry service shouldn't worry about that, the HttpClient is the sender's concern.